mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/DefaultEntityFactory.java

```diff
@@ -230,4 +264,87 @@ private int getClusteringColumnIndex(
       return -1;
     }
   }
+
+  private CqlNameGenerator buildCqlNameGenerator(TypeElement classElement) {
+
+    NamingStrategy namingStrategy = classElement.getAnnotation(NamingStrategy.class);
+    if (namingStrategy == null) {
+      return CqlNameGenerator.DEFAULT;
+    }
+
+    NamingConvention[] conventions = namingStrategy.convention();
+    TypeMirror[] customConverterClasses = readCustomConverterClasses(classElement);
+
+    if (conventions.length > 0 && customConverterClasses.length > 0) {
+      context
+          .getMessager()
+          .error(
+              classElement,
+              "Invalid annotation configuration: %s must have either a 'convention' "
+                  + "or 'customConverterClass' argument, but not both",
+              NamingStrategy.class.getSimpleName());
+      // Return a generator anyway, so that the processor doesn't crash downstream
+      return new CqlNameGenerator(conventions[0]);
```

Comment:
Did we document that preference? (choosing convention over customConverterClass?

Reply:
We only return something here so that the rest of the processor code doesn't crash (it's simpler than trying to abort the whole generation).

The user won't be able to observe this behavior, their build is already failing with an error (from the previous line).